### PR TITLE
Disallow more than two levels of depth to compound namespaces

### LINKED ISSUE #15: Disallow compound namespaces
https://github.com/cs-extended/fig-standards/pull/2#discussion_r41806158

### PR DESCRIPTION
**Pull request by michaelcullum**
- [x] Disallow more than two levels of depth to compound namespaces

Closes #15 

Rationale: Any further depth and it is becoming ugly, probably unnecessary and hard to read whereas this level of depth may be useful. Also see https://github.com/cs-extended/fig-standards/pull/2#discussion_r41806158

### CODE REVIEW

**Comment by michaelcullum** [comment]
@cs-extended/owners

**Comment by samdark** [comment]
Agree.

**Comment by michaelcullum** [comment]
Do we think the wording is clear?

File: proposed/extended-coding-style-guide.md [hunk 1]
```diff
@@ -163,6 +163,19 @@ class FooBar
 
 ```
 
+Compound namespaces with a depth of two or more MUST not be used. Therefore the
+following is the maximum compounding depth allowed:
+```php
+<?php
+
+use Vendor\Package\Namespace\{
```
Comment (samdark) [inline proposed/extended-coding-style-guide.md:171]: Do we need space before `{` here?

File: proposed/extended-coding-style-guide.md [hunk 2]
```diff
@@ -163,6 +163,19 @@ class FooBar
 
 ```
 
+Compound namespaces with a depth of two or more MUST not be used. Therefore the
```
Comment (samdark) [inline proposed/extended-coding-style-guide.md:166]: `Therefore ...` part could be removed. Everything's clear from the first sentence.

File: proposed/extended-coding-style-guide.md (hunk 1 above)
Comment (michaelcullum) [inline proposed/extended-coding-style-guide.md:171]: Nope

File: proposed/extended-coding-style-guide.md (hunk 2 above)
Comment (KorvinSzanto) [inline proposed/extended-coding-style-guide.md:166]: I disagree, I had to ask Michael if this was an anti-example. I think being clear that `SubnamespaceOne\ClassB` is an example of a "compound namespace with a depth of one" is important.

Comment (michaelcullum) [inline proposed/extended-coding-style-guide.md:166]: I'll add this back as I'm leaning on the side of agreeing with @KorvinSzanto

Comment (samdark) [inline proposed/extended-coding-style-guide.md:166]: I don't know. To me it's crystal-clear without it and repetitive with it.

Comment (KorvinSzanto) [inline proposed/extended-coding-style-guide.md:166]: Clear for you, not clear for me. I think we should keep it.

Comment (samdark) [inline proposed/extended-coding-style-guide.md:166]: OK.

Comment (michaelcullum) [inline proposed/extended-coding-style-guide.md:166]: It's not crystal clear to me without this bit and I wrote it....

**Pull request merged**